src/nav/NavigationService.js

```diff
@@ -7,8 +7,13 @@ import type {
   NavigationContainerProps,
 } from 'react-navigation';
 
+/* prettier-ignore */
 const appContainerRef = React.createRef<
-  NavigationContainer<NavigationState, { ... }, NavigationContainerProps<{ ... }, NavigationState>>,
+  React$ElementRef<
```

Comment:
Amazing!! It sounds like we've finally found what `React$ElementRef` is for—thanks! See, e.g., 333c31255af7d90f438c938c74220f1ae4a04eba and its parent c1f9d1543c92b536861085c7d30ef65bdbf0e617; the latter links to https://github.com/zulip/zulip-mobile/pull/4101#discussion_r449259543 (in which I used `React$ElementRef` in very much the wrong way).

Reply:
Hmmmm, I'm actually getting an error when I remove the line

```
// $FlowFixMe - how to tell Flow about `.state`?
```

(below; it seems GitHub won't let me add a comment to a line this far away from a diff chunk)

Reply:
Looks like it's on the thing we're trying to pass as a type argument to `React.createRef`, and it looks familiar:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/nav/NavigationService.js:12:3

Cannot instantiate React.ElementRef because:
 • Either withRouter [1] is not a React component.
 • Or withOptionalNavigationOptions [2] is not a React component.

     src/nav/NavigationService.js
       9│
      10│ /* prettier-ignore */
      11│ const appContainerRef = React.createRef<
      12│   React$ElementRef<
      13│     NavigationContainer<
      14│       NavigationState,
      15│       { ... },
      16│       NavigationContainerProps<{ ... }, NavigationState>>>
      17│ >();
      18│
      19│ const getState = (): NavigationState => {

     flow-typed/npm/react-navigation_v4.x.x.js
 [1] 414│     withRouter<State, Options> &
 [2] 415│     withOptionalNavigationOptions<Options>;
```

Reply:
@gnprice, could you please check and see if you get that error too? I was expecting to be able to remove this `FlowFixMe` just like this WIP commit removes this other fixme:

```
// $FlowFixMe - how to tell Flow about this method?
```

Reply:
(We discussed this [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/decouple.20nav.20from.20redux.20%28.23M3804%29/near/1056167).) It seems like the `React$ElementRef` part is correct, and there's something else going on, probably in the react-navigation libdef. For now the right strategy is probably to type the ref the way we now think is right, and just include whatever fixmes we need to, like this one.